Client/Frontend/Browser/PageZoom/PageZoomView.swift

```diff
@@ -0,0 +1,182 @@
+  @Environment(\.managedObjectContext) private var context
+  
+  private var webView: WKWebView?
+  @State private var minValue = 0.5
+  @State private var maxValue = 3.0
+  @State private var currentValue: Double
+  
+  public static let percentFormatter = NumberFormatter().then {
+    $0.numberStyle = .percent
+    $0.minimumIntegerDigits = 1
+    $0.maximumIntegerDigits = 3
+    $0.maximumFractionDigits = 0
+    $0.minimumFractionDigits = 0
+  }
+  
+  public static let propertyName = "viewScale"
+  public static let notificationName = Notification.Name(rawValue: "com.brave.pagezoom-change")
+  
+  public static let steps = [0.5, 0.75, 0.85,
+                             1.0, 1.15, 1.25,
+                             1.50, 1.75, 2.00,
+                             2.50, 3.0]
+  
+  var dismiss: (() -> Void)?
+  
+  init(webView: WKWebView?) {
+    self.webView = webView
+    
+    // We never re-init, so
+    // it is okay to initialize state here.
+    if let url = webView?.url,
+       let domain = Domain.getPersistedDomain(for: url) {
+      
+      _currentValue = State(initialValue: domain.zoom_level?.doubleValue ?? Preferences.General.defaultPageZoomLevel.value)
+    } else {
+      _currentValue = State(initialValue: webView?.value(forKey: PageZoomView.propertyName) as? Double ?? Preferences.General.defaultPageZoomLevel.value)
+    }
+  }
+  
+  var body: some View {
+    VStack(spacing: 0.0) {
+      Divider()
+      HStack(spacing: 0.0) {
+        Text(Strings.PageZoom.zoomViewText)
+          .font(.system(.subheadline))
+          .frame(maxWidth: .infinity, alignment: .leading)
+        
+        ZoomView(
+          minValue: minValue,
+          maxValue: maxValue,
+          value: $currentValue,
+          onDecrement: decrement,
+          onReset: reset,
+          onIncrement: increment)
+          .frame(maxWidth: .infinity)
+        
+        Button {
+          dismiss?()
+        } label: {
+          Image(systemName: "xmark")
+            .foregroundColor(Color(UIColor.braveBlurple))
+            .font(.system(.footnote).weight(.medium))
+        }
+        .frame(maxWidth: .infinity, alignment: .trailing)
+      }
+      .padding(.horizontal)
+      .padding(.vertical, 6.0)
+    }
+    .background(Color(UIColor.braveBackground))
+  }
+  
+  private func storeChanges() {
+    guard let webView = webView,
+          let url = webView.url else { return }
+    
+    webView.setValue(currentValue, forKey: PageZoomView.propertyName)
+    
+    let domain = Domain.getPersistedDomain(for: url)?.then {
+      $0.zoom_level = currentValue == $0.zoom_level?.doubleValue ? nil : NSNumber(value: currentValue)
+    }
+    
+    try? domain?.managedObjectContext?.save()
+  }
+  
+  private func increment() {
+    guard let index = PageZoomView.steps.firstIndex(of: currentValue) else { return }
+    currentValue = PageZoomView.steps[index + 1]
+    storeChanges()
+  }
+  
+  private func reset() {
+    currentValue = 1.0
```

Comment:
This should be the default value, no?

Reply:
Actually had to think about that for a while. Whether or not resetting meant resetting the page to 100% or whether it reset to the default value. I changed it to default everywhere, so it'll reset to whatever the default value is.
Safari seems to reset to default as well, so it matches now.